### PR TITLE
mimic: core: luminous->(mimic,nautilus): PGMapDigest decode error on luminous end

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/38342

### CODE REVIEW

File: src/messages/MMonMgrReport.h [hunk 1]
```diff
@@ -48,6 +49,23 @@ class MMonMgrReport : public PaxosServiceMessage {
     paxos_encode();
     encode(health_checks, payload);
     encode(service_map_bl, payload);
+
+    if (!HAVE_FEATURE(features, SERVER_NAUTILUS) ||
```
Comment (liewegas) [inline src/messages/MMonMgrReport.h]: only SERVER_MIMIC here

Comment (smithfarm) [reply src/messages/MMonMgrReport.h]: @liewegas Ah, of course! (facepalm) Fixed and repushed.

File: src/messages/MMonMgrReport.h [hunk 2]
```diff
@@ -48,6 +49,22 @@ class MMonMgrReport : public PaxosServiceMessage {
...
+    if (!HAVE_FEATURE(features, SERVER_MIMIC)) {
+      // PGMapDigest had a backwards-incompatible change between
+      // luminous and mimic, and conditionally encodes based on
+      // provided features, so reencode the one in our data payload.
+      // The mgr isn't able to do this at the time the encoded
+      // PGMapDigest is constructed because we don't know which mon we
+      // will target.  Note that this only triggers if the user
+      // upgrades ceph-mgr before ceph-mon (tsk tsk).
+      PGMapDigest digest;
+      auto p = data.cbegin();
```
Comment (liewegas) [inline src/messages/MMonMgrReport.h]: s/cbegin/begin/ so it compiles

**Comment by yuriw** [comment]
[wip-yuri-testing-2019-02-18-1714-mimic](https://trello.com/c/vJf3xQOH/539-wip-yuri-testing-2019-02-18-1714-mimic)

**Comment by yuriw** [comment]
[wip-yuri-testing-2019-02-20-1755-mimic](https://trello.com/c/G3p1tb0I/543-wip-yuri-testing-2019-02-20-1755-mimic)

**Comment by yuriw** [comment]
needs retest after https://github.com/ceph/ceph/pull/26451 merged and backported

**Comment by neha-ojha** [comment]
@yuriw you mean after https://github.com/ceph/ceph/pull/26636 is merged?

**Comment by yuriw** [comment]
@neha-ojha yes corrected typo, thx

**Comment by smithfarm** [comment]
Changelog

* rebased
* cherry-picked Sage's follow-up fix from https://github.com/ceph/ceph/pull/26636

**Comment by yuriw** [comment]
[wip-PR2645-mimic](https://trello.com/c/KN46jN5q/549-wip-pr2645-mimic)

**Pull request merged**